nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-framework-cluster/src/main/java/org/apache/nifi/cluster/coordination/http/endpoints/StatusHistoryEndpointMerger.java

```diff
@@ -109,13 +119,49 @@ public NodeResponse merge(URI uri, String method, Set<NodeResponse> successfulRe
                 noReadPermissionsComponentDetails = nodeStatus.getComponentDetails();
             }
 
+            if (!nodeStatus.isIncludeCounters()) {
```

Comment:
I'm not sure we need to add a new field to the `nodeStatus` as the read permission is already present in the corresponding `nodeResponseEntity`.